src/sql/src/plan/statement/ddl.rs

```diff
@@ -3483,14 +3483,25 @@ pub fn plan_create_connection(
             let c = AwsPrivatelinkConnectionOptionExtracted::try_from(with_options)?;
             let connection = AwsPrivatelinkConnection::try_from(c)?;
             if let Some(supported_azs) = scx.catalog.aws_privatelink_availability_zones() {
+                let mut unique_azs: BTreeSet<String> = BTreeSet::new();
+                let mut duplicate_azs: BTreeSet<String> = BTreeSet::new();
+                // Validate each AZ is supported
                 for connection_az in &connection.availability_zones {
+                    if unique_azs.contains(connection_az) {
+                        duplicate_azs.insert(connection_az.to_string());
+                    } else {
+                        unique_azs.insert(connection_az.to_string());
+                    }
                     if !supported_azs.contains(connection_az) {
                         return Err(PlanError::InvalidPrivatelinkAvailabilityZone {
                             name: connection_az.to_string(),
                             supported_azs,
                         });
                     }
                 }
+                if duplicate_azs.len() > 0 {
+                    return Err(PlanError::DuplicatePrivatelinkAvailabilityZone { duplicate_azs });
+                }
```

Comment:
There's a handy `duplicates()` method in `itertools::Itertools` that should work perfectly here!

```suggestion
                for connection_az in &connection.availability_zones {
                    if !supported_azs.contains(connection_az) {
                        return Err(PlanError::InvalidPrivatelinkAvailabilityZone {
                            name: connection_az.to_string(),
                            supported_azs,
                        });
                    }
                }
                let duplicate_azs = connection.availability_zones.iter().duplicates().map(|az| az.to_string()).collect();
                if !duplicate_azs.is_empty() {
                    return Err(PlanError::DuplicatePrivatelinkAvailabilityZone { duplicate_azs });
                }
```